WalletWasabi/WabiSabi/Backend/Banning/CoinVerifierAuditArchiver.cs

```diff
@@ -0,0 +1,61 @@
+using NBitcoin;
+using Nito.AsyncEx;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.WabiSabi.Backend.Banning;
+
+public class CoinVerifierAuditArchiver
+{
+	public CoinVerifierAuditArchiver(string directoryPath)
+	{
+		IoHelpers.EnsureDirectoryExists(directoryPath);
+		BaseDirectoryPath = directoryPath;
+	}
+
+	private string BaseDirectoryPath { get; }
+
+	private AsyncLock FileAsyncLock { get; } = new();
+
+	public async Task SaveAuditsAsync(IEnumerable<CoinVerifyResult> results, CancellationToken cancellationToken)
+	{
+		StringBuilder fileContent = new();
```

Comment:
Just create a list of strings and let `File.AppendAllLinesAsync` does the line stuff.